### PR TITLE
Task/pytorch conf

### PR DESCRIPTION
**Pull request by vanpelt**

### CODE REVIEW

File: wandb/data_types.py [hunk 1]
```diff
@@ -115,18 +158,359 @@ def from_keras(cls, model):
...
+        torch = util.get_module("torch", "Could not import torch")
+        hooks = []
+        modules = set()
+        layers = 0
+        graph = self
+        if criterion:
+            graph.criterion = criterion
+            graph.criterion_passed = True
+
+        for name, sub_module in module._modules.items():
```
Comment (adrianbg) [inline wandb/data_types.py]: Is this better than `module.named_children()`? Maybe works across versions?

File: wandb/data_types.py [hunk 2]
```diff
@@ -115,18 +158,359 @@ def from_keras(cls, model):
...
+            name = name or str(layers)
+            if prefix:
+                name = prefix + "." + name
+            layers += 1
+            if not isinstance(sub_module, torch.nn.Module):
+                # TODO: Why does this happen?
```
Comment (adrianbg) [inline wandb/data_types.py]: Hmm... maybe there's a better superclass to use. Is it maybe the built-in modules that break this?

File: wandb/wandb_torch.py [hunk 3]
```diff
@@ -41,31 +49,49 @@ def forward(self, x):
...
             if name is None:
                 raise wandb.Error('Need a name to log stats for a Variable.')
             var = variable_or_module
             if values:
-                self.log_tensor_stats(var.data, prefix + name)
+                self.log_tensor_stats(var.data, 'parameters/' + prefix + name)
             if gradients:
                 self._hook_variable_gradient_stats(
-                    var, prefix + name + ':grad')
+                    var, 'gradients/' + prefix + name)
```
Comment (adrianbg) [inline wandb/wandb_torch.py]: What about doing '/grad' so gradients are grouped with their respective parameters?

File: install-ml.py [hunk 4]
```diff
@@ -4,7 +4,9 @@
...
 version = "".join([str(v) for v in sys.version_info[:2]])
 if version == "27":
-    call(["pip", "install", "http://download.pytorch.org/whl/cpu/torch-0.4.0-cp27-cp27mu-linux_x86_64.whl"])
+    call(["pip", "install", "http://download.pytorch.org/whl/cpu/torch-0.4.1-cp27-cp27mu-linux_x86_64.whl"])
+    call(["pip", "install", "torchvision"])
 else:
-    call(["pip", "install",
-          "http://download.pytorch.org/whl/cpu/torch-0.4.0-cp{version}-cp{version}m-linux_x86_64.whl".format(version=version)])
+    call(["pip", "install", "torch_nightly", "-f",
+          "https://download.pytorch.org/whl/nightly/cpu/torch_nightly.html"])
```
[the remark this answers is not in the document]
Comment (vanpelt) [reply install-ml.py]: This way tests are run against pytorch 1 which I assume many of our users would be doing.  For python2 tests I'm running stable.

File: wandb/data_types.py [hunk 5]
```diff
@@ -61,8 +67,45 @@ def to_json(payload, mode="history"):
...
+
+    def add_node(self, node=None, **node_kwargs):
+        if node is None:
+            node = Node(**node_kwargs)
+        elif node_kwargs:
+            raise ValueError('Only pass one of either node ({node}) or other keyword arguments ({node_kwargs})'.format(
+                node=node, node_kwargs=node_kwargs))
+        self.nodes.append(node)
+        # TODO: Adrian, why was this here?
+        #assert node.id not in self.nodes_by_id
```
[the remark this answers is not in the document]
Comment (vanpelt) [reply wandb/data_types.py]: This breaks the keras callback tests so I just removed it.

File: wandb/data_types.py (hunk 1 above)
Comment (vanpelt) [reply wandb/data_types.py]: Yeah, I took this from an example I found online, I'll add a comment.
